src/System.Devices.Gpio/System.Devices.Gpio/GPIOPin.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace System.Devices.Gpio
+{
+    public enum GpioPinMode
+    {
+        Input = 0,
+        Output = 1
+    }
+
+    public enum GpioPinValue
+    {
+        Low = 0,
+        High = 1
+    }
+
+    public enum GpioScheme
+    {
+        Board,
+        BCM
+    }
+
+    public class GpioPin
+    {
+        protected GpioDriver Driver;
+
+        public int Number { get; }
+
+        public GpioPinMode Mode
+        {
+            get
+            {
+                return Driver.GetPinMode(Number);
+            }
+            set
+            {
+                Driver.SetPinMode(Number, value);
+            }
+        }
+
+        public GpioPin(GpioDriver driver, GpioScheme numberKind, int number, GpioPinMode mode)
```

Comment:
Can we come up with a better name than "numberKind"?  How about "mode". 

What's called GpioPinMode right now feels like System.IO.FileAccess enum (Read, ReadWrite, Write).  Perhaps this should be a PinAccess enum.